cppcache/integration-test/QueryHelper.hpp

```diff
@@ -315,76 +315,58 @@ void QueryHelper::getPDXObject(std::shared_ptr<Region>& rptr) {
 
 bool QueryHelper::verifyRS(std::shared_ptr<SelectResults>& resultSet,
                            size_t expectedRows) {
-  if (!std::dynamic_pointer_cast<ResultSet>(resultSet)) {
-    return false;
-  }
-
-  auto rsptr = std::static_pointer_cast<ResultSet>(resultSet);
-
-  size_t foundRows = 0;
-
-  SelectResultsIterator iter = rsptr->getIterator();
+  if (auto rsptr = std::static_pointer_cast<ResultSet>(resultSet)) {
+    size_t foundRows = 0;
+    for (auto&& row : *rsptr) {
+      foundRows++;
+    }
 
-  for (size_t rows = 0; rows < rsptr->size(); rows++) {
-    auto ser = (*rsptr)[rows];
-    foundRows++;
+    printf("found rows %zd, expected %zd \n", foundRows, expectedRows);
+    if (foundRows == expectedRows) return true;
   }
-
-  printf("found rows %zd, expected %zd \n", foundRows, expectedRows);
-  if (foundRows == expectedRows) return true;
-
   return false;
 }
 
 bool QueryHelper::verifySS(std::shared_ptr<SelectResults>& structSet,
                            size_t expectedRows, int32_t expectedFields) {
-  if (!std::dynamic_pointer_cast<StructSet>(structSet)) {
-    if (expectedRows == 0 && expectedFields == 0) {
-      return true;  // quite possible we got a null set back.
-    }
-    printf("we have structSet itself nullptr \n");
-    return false;
-  }
-
-  auto ssptr = std::static_pointer_cast<StructSet>(structSet);
+  if (auto ssptr = std::static_pointer_cast<StructSet>(structSet)) {
+    size_t foundRows = 0;
+    for (auto&& ser : *ssptr) {
+      foundRows++;
 
-  size_t foundRows = 0;
+      auto siptr = std::dynamic_pointer_cast<Struct>(ser);
 
-  for (SelectResults::Iterator iter = ssptr->begin(); iter != ssptr->end();
-       iter++) {
-    auto ser = (*iter);
-    foundRows++;
+      if (siptr == nullptr) {
+        printf("siptr is nullptr \n\n");
```

Comment:
cout?

Reply:
Not changing existing test code like that.